onnxruntime/test/providers/cpu/tensor/tile_op_test.cc

```diff
@@ -8,210 +8,223 @@ namespace onnxruntime {
 namespace test {
 
 template <typename T>
-void RunTest(std::initializer_list<T> input,
-             std::initializer_list<int64_t> input_dims,
-             std::initializer_list<int64_t> repeat,
-             std::initializer_list<int64_t> repeat_dims,
-             std::initializer_list<T> output,
-             std::initializer_list<int64_t> output_dims) {
+std::vector<T> InputData(size_t size) {
+  std::vector<T> result(size);
+  for (size_t i = 0; i < size; i++) {
+    result[i] = static_cast<T>(i);
+  }
+  return result;
+}
+
+template <>
+std::vector<MLFloat16> InputData<MLFloat16>(size_t size) {
+  std::vector<MLFloat16> result(size);
+  for (size_t i = 0; i < size; i++) {
+    result[i] = MLFloat16(static_cast<float>(i));
+  }
+  return result;
+}
+
+template <typename T>
+void RunTest(const std::vector<int64_t>& input_dims, const std::vector<int64_t>& repeats) {
+  size_t input_size =
+      static_cast<size_t>(std::accumulate(input_dims.begin(), input_dims.end(), 1LL, std::multiplies<int64_t>()));
+  std::vector<T> input_data = InputData<T>(input_size);
+  size_t rank = input_dims.size();
+  std::vector<int64_t> repeats_dims(1);
+  repeats_dims[0] = static_cast<int64_t>(rank);
+  std::vector<int64_t> output_dims(rank);
+  for (size_t i = 0; i < rank; ++i) {
+    output_dims[i] = input_dims[i] * repeats[i];
+  }
+  size_t output_size =
+      static_cast<size_t>(std::accumulate(output_dims.begin(), output_dims.end(), 1LL, std::multiplies<int64_t>()));
+  std::vector<T> output_data(output_size);
+  std::vector<int64_t> input_strides(rank);
+  std::vector<int64_t> output_strides(rank);
+  input_strides[rank - 1] = output_strides[rank - 1] = 1;
+  if (rank > 1) {
+    for (size_t i = rank - 2;; --i) {
+      input_strides[i] = input_dims[i + 1] * input_strides[i + 1];
+      output_strides[i] = output_dims[i + 1] * output_strides[i + 1];
+      if (i == 0) break;
```

Comment:
>if (i == 0) break; [](http://example.com/codeflow?start=6&length=18)

nit: why not put it in the loop condition?

Reply:
because i is size_t and will always >= 0.

Reply:
😳